emote/sac.py

```diff
@@ -295,6 +300,71 @@ def load_state_dict(self, state_dict: Dict[str, Any]):
         super().load_state_dict(state_dict)
 
 
+class LoggingProxyWrapper:
+    def __init__(self, inner: AgentProxy, cycle: int):
+        self._inner = inner
+        self._cycle = cycle
+
+        self._counter = 0
+
+    def __call__(
+        self,
+        observations: Dict[AgentId, DictObservation],
+    ) -> Dict[AgentId, DictResponse]:
+        self._counter += 1
+
+        if (self._counter % self._cycle) == 0:
+            self._end_cycle()
+            self._counter = 0
+
+        return self._inner(observations)
+
+
+class AgentProxyOnnxExporter(OnnxGeneratorMixin, OnnxStorage, LoggingProxyWrapper):
```

Comment:
I don't understand what we gain from having the Onnx-things being mixins. They are not used anywhere else. Why not just let them be private methods?